meshtastic/node.py

```
@@ -998,33 +1127,160 @@ def setURL(self, url: str, addOnly: bool = False) -> None:
-                self.writeChannel(channel_index)
+                rollback_failed = False
+                written_index_set = set(written_indices)
+                for index, rollback_channel in original_channels_by_index.items():
+                    if index in written_index_set:
+                        try:
+                            self._write_channel_snapshot(
+                                rollback_channel,
+                                adminIndex=admin_index_for_write,
+                            )
+                        # Best-effort rollback path; keep attempting remaining steps.
+                        except Exception:
```

Ruff (0.15.5)

[warning] Do not catch blind exception: `Exception`

(BLE001)